Copyediting of CrateDB docs "Best Practices" section (#176)
crate/crate-howtos: docs/best-practices/data-import.rst

:replicas:
  1

> We recommend you choose the number of shards wisely. It depends on the number

Reviewer: unclear antecedent. perhaps "This choice depends" or similar?

of nodes in the cluster as well as on the amount of data that goes into the
table.

.. CAUTION::

  Be careful, you cannot change the number of shards once the table is created!

> Assuming there are 6 nodes in the cluster, we put 2 shards on each node,

Reviewer: "six" instead of "6" per https://oreillymedia.github.io/production-resources/styleguide/#dates_and_numbers

same with "2"

Author: "unless the same object appears in a sentence with an object 10 or over": in this case "12 shards". Otherwise you end up with "six nodes, two shards, 12 shards", which looks odd.

Reviewer: ah good catch thx

> giving us a total of 12 shards. This should be enough for millions of

Reviewer: "us" makes me wonder if it's a big issue to also correct the use of pronouns in this doc to match https://developers.google.com/style/pronouns (no worries if not!)

Author: Yup! I think there were only a few 'wes'

   {"id": 1, "name": "foo", "day_joined": 1408312800, "bio": "Lorem ipsum dolor sit amet, consectetuer adipiscing elit.", "address": {"city": "Dornbirn", "country": "Austria"}}
   {"id": 2, "name": "bar", "day_joined": 1408312800, "bio": "Lorem ipsum dolor sit amet, consectetuer adipiscing elit.", "address": {"city": "Berlin", "country": "Germany"}}

``COPY FROM``
-------------

Upon execution, each node will check the provided path *locally* to see whether

> the file exists and to import the data it contains. Consequently this command

Reviewer: ```suggestion
the file exists and to import the data it contains. Consequently, this command
```

will check ``/tmp/best_practice_data/`` on each node in the cluster to import

> data from a file called 'users.json'. Please note that if the file is not

Reviewer: ```suggestion
data from a file called ``users.json``. Please note that if the file is not
```

found, the command will return successfully, reporting

> ``COPY OK, 0 rows affected (... sec)``.

Reviewer: this literal can be wrapped onto the prev line

.. NOTE::

  When importing data using ``COPY FROM``, CrateDB does not check whether the
  types from the columns and the types from the import file match. It does not
  cast the types to their target but will always import the data as in the

>   source file(s).

Reviewer: "(s)" strikes me as bad style

(I realize you didn't change anything to do with the wording here but it jumped out at me)

``bulk_size``. Setting ``bulk_size`` too high might consume a lot of node
resources while a low ``bulk_size`` can increase the overhead per request.

Compression
...........

If you do not have your data locally to the nodes, but somewhere on the

> network, e.g. on a NAS or on ``S3``, it's recommended to use ``gzip``

Reviewer: ```suggestion
network, e.g., on a NAS or on ``S3``, it's recommended to use ``gzip``
```

Author: That makes too many commas. I'm going to split it then

  COPY OK, 23 rows affected (... sec)

This way, CrateDB does not need to resolve the partition for each row that is
imported, but can store it directly into the correct place resulting in a much
faster import.

However, it's still possible (but not recommended) to import into partitioned
tables without the ``PARTITION`` clause and have the column value in the
source.

When importing data into a partitioned table with existing partitions, it may

> be desirable to apply import optimizations, like e.g. to disable the `Refresh

Reviewer: "like e.g." seems inelegant

Author: Yeah this sentence needs a rewrite anyway

  - Table creation of `PARTITIONED BY clause`_
  - `Alter a partitioned table`_

Summary
=======

To sum up the points described above, importing huge datasets is not difficult

> as long as a few things are kept in mind. These are:

Reviewer: perhaps "providing" is easier for non-native speakers?

Author: I'd think that's less easy! Compare Dutch: "zolang", German: "solange", Spanish: "siempre y cuando", etc...

Reviewer: okay!
